### ISSUE TITLE
Schema validation does not see the _ at the start of t_strings

### ISSUE THREAD

**Issue by stevecotton** [wesnoth/wesnoth#4537]
### Game and System Information
Linux, current git master at 1a67d53e7dd22c675e1fcadf963c7474035e1083
Running the validator with `build/debug/wesnoth --userconfig-dir ~/.config/wesnoth1.15/ --data-dir $PWD --validate data/_main.cfg --preprocess-defines=TEST,SCHEMA_SHOULD_SKIP_THIS`
### Describe the bug
The engine for schema validation strips the underscore from translatable strings before checking the value with the schema, which means that the schema can't enforce translatability of strings.

The schema declares that t_strings should <i>optionally</i> start with an underscore; this bug is hidden because the schema says the underscore is only optional:
https://github.com/wesnoth/wesnoth/blob/1a67d53e7dd22c675e1fcadf963c7474035e1083/data/schema/types/basic.cfg#L31-L34

I want to change that regexp is to only allowing (empty without underscore) or (value starting with underscore), `value="()|(_.*)"`. However, doing that will return a huge list of errors, finishing with
```
error validation: Invalid value 'Teal' in key 'name=' in tag [color_range]
at core/team-colors.cfg:287
    included from game_config.cfg:123
    included from data/_main.cfg:102
```
even though there is an underscore on that line (and the other lines that are complained about):
https://github.com/wesnoth/wesnoth/blob/1a67d53e7dd22c675e1fcadf963c7474035e1083/data/core/team-colors.cfg#L287

**Comment by CelticMinstrel** [comment]
This is probably not caused by anything in the schema; it's likely due to the WML parser converting the raw WML into actual typed data by the time the schema sees it. I think the idea of trying to match translatable strings via regex is probably flawed to begin with, given that. But I'm not sure if there's really a better way, other than maybe accepting a `translatable=yes` key in the `[type]` tag…

Another thing to note is that not all strings that _could_ be translatable are marked as such. For example, I believe most of the test scenarios do not mark their strings translatable.

**Comment by soliton-** [comment]
I think it'd be fine for `t_string` to be a type just for documentation purposes. As @CelticMinstrel says we don't want to enforce marking every string translatable just because the WML key supports it.

I'd change the regex to: .*

**Comment by stevecotton** [comment]
Implemented in #5800.
